lib/libc/minimal/source/stdout/prf.c

```diff
@@ -771,10 +766,9 @@ int z_prf(int (*func)(), void *dest, const char *format, va_list vargs)
 					c = width;
 				}
 
-				for (cptr = buf; c > 0; c--, cptr++, count++) {
-					if ((*func)(*cptr, dest) == EOF) {
-						return EOF;
-					}
+				count += c;
+				for (cptr = buf; c > 0; c--, cptr++) {
+					PUTC(*cptr);
 				}
```

Comment:
`#undef PUTC` here otherwise this local macro will be available out of the scope of this function